v3/mqtt.go

```diff
@@ -465,91 +577,171 @@ func (conn *mqttConn) runPacketWriter(wg *sync.WaitGroup) {
 		wg.Done()
 	}()
 
-	// Run until closed
-	for p := range conn.outPktCh {
-		err := conn.writePacket(p)
-		if err != nil {
-			logError("Unable to write out packet %s", p.Type())
-			switch p.Type() {
-			case packet.PINGREQ:
-				conn.pingRespCh <- false
-			case packet.SUBSCRIBE:
-				subPkt := p.(*packet.SubscribePacket)
-				conn.queueAckCh <- MqttQueueResult{subPkt.PacketID, err}
-			case packet.PUBLISH:
-				pubPkt := p.(*packet.PublishPacket)
-				conn.queueAckCh <- MqttQueueResult{pubPkt.PacketID, err}
-			default:
-				// We don't have a packet ID (or did not expect to receive this
-				// packet type and fail in sending it
-				err := fmt.Errorf("Failed to queue packet %s", p.Type())
-				conn.queueAckCh <- MqttQueueResult{0, err}
+	shouldLoop := true
```

Comment:
How about use a label such as the following?
```suggestion
LOOP:
```

Reply:
I have removed that code so the label is unnecessary.